tests/ctests/IsdTests.cpp

```diff
@@ -0,0 +1,679 @@
+
+  ale::Isd isd(json_str);
+
+  ASSERT_EQ(isd.usgscsm_name_model, "USGS_ASTRO_LINE_SCANNER_SENSOR_MODEL");
+  ASSERT_EQ(isd.name_platform, "MARS_RECONNAISSANCE_ORBITER");
+  ASSERT_EQ(isd.image_id, "TEST_IMAGE");
+  ASSERT_EQ(isd.name_sensor, "CONTEXT CAMERA");
+  ASSERT_DOUBLE_EQ(isd.semi_major, 3396.19);
+  ASSERT_DOUBLE_EQ(isd.semi_minor, 3376.2);
+
+  ASSERT_EQ(isd.detector_sample_summing, 1);
+  ASSERT_EQ(isd.detector_line_summing, 1);
+
+  ASSERT_DOUBLE_EQ(isd.focal_length, 352.9271664);
+  ASSERT_DOUBLE_EQ(isd.focal_uncertainty, 1);
+  ASSERT_DOUBLE_EQ(isd.detector_center_line, 0.430442527);
+  ASSERT_DOUBLE_EQ(isd.detector_center_sample,  2542.96099);
+
+  ASSERT_DOUBLE_EQ(isd.starting_detector_line, 0);
+  ASSERT_DOUBLE_EQ(isd.starting_detector_sample, 0);
+
+  ASSERT_DOUBLE_VECTOR_EQ(isd.focal2pixel_line, std::vector<double>({0.0, 142.85714285714, 0.0}));
+  ASSERT_DOUBLE_VECTOR_EQ(isd.focal2pixel_sample, std::vector<double>({0.0, 0.0, 142.85714285714}));
+
+  ASSERT_EQ(isd.distortion_model, ale::DistortionType::RADIAL);
+  ASSERT_DOUBLE_VECTOR_EQ(isd.distortion_coefficients, std::vector<double>({-0.0073433925920054505, 2.8375878636241697e-05, 1.2841989124027099e-08}));
+
+  ASSERT_EQ(isd.image_lines, 400);
+  ASSERT_EQ(isd.image_samples, 5056);
+
+  ASSERT_DOUBLE_EQ(isd.max_reference_height, 1000);
+  ASSERT_DOUBLE_EQ(isd.min_reference_height, -1000);
+
+  ASSERT_DOUBLE_2D_VECTOR_EQ(isd.line_scan_rate, std::vector<std::vector<double>>({std::vector<double>({0.5, -0.37540000677108765, 0.001877})}));
+
+  ASSERT_DOUBLE_EQ(isd.starting_ephemeris_time, 297088762.24158406);
+  ASSERT_DOUBLE_EQ(isd.center_ephemeris_time, 297088762.61698407);
+}
+
+TEST(Isd, LogFile) {
+  ale::json j;
+  j["log_file"] = "fake/path";
+  EXPECT_STREQ(ale::getLogFile(j).c_str(), "fake/path");
+}
+
+TEST(Isd, TransverseDistortion) {
+  ale::json trans;
+  trans["optical_distortion"]["transverse"]["x"] = {1};
+  trans["optical_distortion"]["transverse"]["y"] = {2};
+
+  std::vector<double> coeffs = ale::getDistortionCoeffs(trans);
+  EXPECT_EQ(ale::getDistortionModel(trans), ale::DistortionType::TRANSVERSE);
+  EXPECT_DOUBLE_EQ(coeffs[0], 1);
+  EXPECT_DOUBLE_EQ(coeffs[10], 2);
```

Comment:
Before accessing anything at an index, you should assert the vector size is what you expect.